satpy/readers/seviri_l2_grib.py

```diff
@@ -0,0 +1,265 @@
+    https://www.wmo.int/pages/prog/www/WMOCodes/Guides/GRIB/GRIB2_062006.pdf
+
+"""
+
+import logging
+import numpy as np
+import xarray as xr
+import dask.array as da
+import eccodes as ec
+
+from datetime import timedelta
+
+from satpy.readers.file_handlers import BaseFileHandler
+from satpy.readers._geos_area import get_area_definition
+from satpy.readers.seviri_base import (calculate_area_extent,
+                                       PLATFORM_DICT,
+                                       REPEAT_CYCLE_DURATION)
+from satpy import CHUNK_SIZE
+
+logger = logging.getLogger(__name__)
+
+
+class SeviriL2GribFileHandler(BaseFileHandler):
+    """Reader class for SEVIRI L2 products in GRIB format."""
+
+    def __init__(self, filename, filename_info, filetype_info):
+        """Read the global attributes and prepare for dataset reading."""
+        super().__init__(filename, filename_info, filetype_info)
+        # Turn on support for multiple fields in single GRIB messages (required for SEVIRI L2 files)
+        ec.codes_grib_multi_support_on()
+        self._read_global_attributes()
+
+    def _read_global_attributes(self):
+        """Read the global product attributes from the first message.
+
+        Read the information about the date and time of the data product,
+        the projection and area definition and the number of messages.
+
+        """
+        with open(self.filename, 'rb') as fh:
+            gid = ec.codes_grib_new_from_file(fh)
+
+            if gid is None:
+                # Could not obtain a valid message id: set attributes to None, number of messages to 0
+                logger.warning("Could not obtain a valid message id in GRIB file")
+
+                self._ssp_lon = None
+                self._nrows = None
+                self._ncols = None
+                self._pdict, self._area_dict = None, None
+
+                return
+
+            # Read SSP and date/time
+            self._ssp_lon = self._get_from_msg(gid, 'latitudeOfSubSatellitePointInDegrees')
+
+            # Read number of points on the x and y axes
+            self._nrows = self._get_from_msg(gid, 'Ny')
+            self._ncols = self._get_from_msg(gid, 'Nx')
+
+            # Creates the projection and area dictionaries
+            self._pdict, self._area_dict = self._get_proj_area(gid)
+
+            # Determine the number of messages in the product by iterating until an invalid id is obtained
+            i = 1
+            ec.codes_release(gid)
+            while True:
+                gid = ec.codes_grib_new_from_file(fh)
+                if gid is None:
+                    break
+                ec.codes_release(gid)
+                i = i+1
+
+    @property
+    def start_time(self):
+        """Return the sensing start time."""
+        return self.filename_info['start_time']
+
+    @property
+    def end_time(self):
+        """Return the sensing end time."""
+        return self.start_time + timedelta(minutes=REPEAT_CYCLE_DURATION)
```

Comment:
Isn't the actual scanning duration 12:30 minutes?

Reply:
approximately, but this actual scanning time is not considered by any of the SEVIRI readers. all readers assume 15 repeat cycle duration for the end time. i suggest we leave this as is.

Reply:
Ok, agree!